libs/metrics/social_distancing.py

```diff
@@ -0,0 +1,187 @@
+            # Append social distancing violations
+            objects_logs[row_hour][d["tracking_id"]]["distance_violations"].append(
+                {
+                    "time": row_time,
+                    "infrigement": index in ast.literal_eval(csv_row["ViolationsIndexes"])
+                }
+            )
+
+    @classmethod
+    def generate_hourly_metric_data(cls, objects_logs):
+        summary = np.zeros((len(objects_logs), 5), dtype=np.long)
+        for index, hour in enumerate(sorted(objects_logs)):
+            hour_objects_detections = objects_logs[hour]
+            for detection_object in hour_objects_detections.values():
+                summary[index] += cls.process_distance_violation_for_object(
+                    detection_object["distance_violations"])
+        return summary
+
+    @classmethod
+    def process_distance_violation_for_object(cls, distance_violations: List[dict]) -> Tuple[int, int]:
+        """
+        Receives a list with the "social distancing detections" (for a single person) and returns a
+        tuple with the summary of detections and violations (grouped by severity). Consecutive detections in
+        the same state are grouped and returned as a single one. Detections lower than the constant
+        PROCESSING_COUNT_THRESHOLD are ignored.
+
+        The infrigement categories are :
+            - Low: Between 10 seconds 30 seconds
+            - High: Between 30 and 60 seconds
+            - Critical: More than 60 seconds
+        """
+        # TODO: The categories values defined need to be updated taking into account the OMS recommendations.
+        # The current values only have demo purposes
+        current_status = None
+        processing_status = None
+        processing_count = 0
+        processing_time = None
+
+        CRITICAL_THRESHOLD = 60
+        HIGH_THRESHOLD = 30
+        LOW_TRESHOLD = 10
+
+        detections = []
+        if distance_violations:
+            for dist_violation in distance_violations:
+                status = dist_violation["infrigement"]
+                if processing_status != status:
+                    processing_status = status
+                    processing_time = dist_violation["time"]
+                    processing_count = 0
+                processing_count += 1
+                if current_status != processing_status and processing_count >= cls.processing_count_threshold:
+                    # Object was enouth time in the same state, change it
+                    if current_status is not None:
+                        # Append the previous status in the detections list
+                        seconds_in_status = (dist_violation["time"] - processing_time).seconds
+                        detections.append({"status": status, "seconds": seconds_in_status})
+                    current_status = processing_status
+                    processing_time = dist_violation["time"]
```

Comment:
I think this line should be in the if

Reply:
If we put the processing time inside the if we will lose the `processing_time` of the first status.

Reply:
I mean like this you are overriding the `processing_time` so you will lose the time that elapses from the status change until `processing_count >= cls.processing_count_threshold`, right?

Reply:
:thinking: You are right, but inside the if I will lose the first status.
I need to use a new variable to store the time when the status changes.

Reply:
I just committed a fix.